TypeCobol/Compiler/Types/GroupType.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.IO;
+using TypeCobol.Compiler.Scopes;
+using TypeCobol.Compiler.Symbols;
+
+using static TypeCobol.Compiler.Symbols.Symbol;
+
+namespace TypeCobol.Compiler.Types
+{
+    /// <summary>
+    /// Class that represents a group type
+    /// </summary>
+    public class GroupType : Type
+    {
+        /// <summary>
+        /// The fields of this GroupType.
+        /// </summary>
+        public Domain<VariableSymbol> Fields { get; internal set; }
+
+        /// <summary>
+        /// Scope Owner constructor
+        /// </summary>
+        /// <param name="owner">Owner of the group scope if any</param>
+        public GroupType(Symbol owner)
+            : base(Tags.Group)
+        {
+            Fields = new Domain<VariableSymbol>(owner);
+        }
+
+        internal override void SetFlag(Flags flag, bool value, bool propagate = false)
+        {
+            base.SetFlag(flag, value, propagate);
+            if (propagate)
+            {
+                foreach (var varSym in Fields)
+                {
+                    varSym.SetFlag(flag, value, true);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Using Cobol some records can have a leading type
+        /// which can be a USAGE type or a PICTURE Type, for instance;
+        /// 
+        /// 10  checkToDo-value PIC X VALUE LOW-VALUE.
```

Comment:
I suggest to rename LeadingType to UsageType then and update the comment.

Reply:
But wait, if it's an elementary data description item, how should the data conditions be created and attached to their parent ?

Reply:
That's a good question.
I think VariableSymbol should know their children Symbol.

Reply:
If we really want to go deep in type comparison, then DataCondition and theirs values should also be compared.

Reply:
Like:
```cobol
01 var1 pic X.
   88 var1-valid value 'A' 'B'.
   88 var1-A value 'A'.
   88 var1-B value 'B'.

01 var2 pic X.
   88 var2-C value 'C'.

move var1 to var2
```
In this case, one could warn that var1 and var2 don't share same DataCondition.
In this PR, we can discuss on how to represent this with Symbol/SymbolType.
But the actual check/warning is not the subject of this PR.

Reply:
But in this case the type of the VariableSymbol instance will be a GroupType thus children will be the Fields of the GroupType

Reply:
The SemanticDomain uses Nodes to build symbol hierarchy. As soon as a DataDefinition node has children, it will be translated into a VariableSymbol with GroupType. In that regard, GroupType is not strictly equivalent to group item in Cobol specs because of level-88.
This model is fine but we'll have to be careful later with type compatibility because a GroupType may be compatible with PictureType and that's where the LeadingType will be used.

Reply:
See PR #1723, it includes the new property `IsElementary`.